extension/src/tree/errorDecorationProvider.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  CancellationToken,
+  Event,
+  EventEmitter,
+  FileDecoration,
+  FileDecorationProvider,
+  ProviderResult,
+  ThemeColor,
+  Uri,
+  window
+} from 'vscode'
+import { DecoratableTreeItemScheme } from '.'
+import { Disposable } from '../class/dispose'
+
+export abstract class ErrorDecorationProvider
```

Comment:
This is the base class that I mentioned [here](https://github.com/iterative/vscode-dvc/pull/2301/files#r961303414).